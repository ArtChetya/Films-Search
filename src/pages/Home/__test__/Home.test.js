import { shallow } from 'enzyme/build';
import React from 'react';
import { Home } from '..';

describe('Home component', () => {
    let sortByOptions;
    let sortById;
    let setSortById;
    let onSearch;
    let isLoading;

    beforeAll(() => {
        sortByOptions = [
            { id: 'release_date', label: 'release date' },
            { id: 'vote_average', label: 'rating' }
        ];

        sortById = 'release_date';

        setSortById = jest.fn();

        onSearch = jest.fn();

        isLoading = false;
    });

    it('should be rendered correctly', () => {
        const component = shallow(
            <Home
                sortByOptions={sortByOptions}
                sortById={sortById}
                setSortById={setSortById}
                onSearch={onSearch}
                isLoading={isLoading}
            />
        );
        expect(component).toMatchSnapshot();
    });

    it('should be rendered correctly with loading', () => {
        const component = shallow(
            <Home
                sortByOptions={sortByOptions}
                sortById={sortById}
                setSortById={setSortById}
                onSearch={onSearch}
                isLoading
            />
        );
        expect(component).toMatchSnapshot();
    });
});
