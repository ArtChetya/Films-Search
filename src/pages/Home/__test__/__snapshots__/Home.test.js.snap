// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should be rendered correctly 1`] = `
<PageGrid
  container={true}
  direction="column"
  wrap="nowrap"
>
  <Header
    content={
      <SearchForm
        onSearch={[MockFunction]}
      />
    }
    hasSearch={false}
  />
  <SplitPane
    left={<Connect(FilmsFound) />}
    right={
      <SortBy
        onOptionChange={[MockFunction]}
        options={
          Array [
            Object {
              "id": "release_date",
              "label": "release date",
            },
            Object {
              "id": "vote_average",
              "label": "rating",
            },
          ]
        }
        selectedOptionId="release_date"
      />
    }
  />
  <PageContentWrapper>
    <Connect(FilmsList) />
  </PageContentWrapper>
  <Footer />
</PageGrid>
`;

exports[`Home component should be rendered correctly with loading 1`] = `
<PageGrid
  container={true}
  direction="column"
  wrap="nowrap"
>
  <Header
    content={
      <SearchForm
        onSearch={[MockFunction]}
      />
    }
    hasSearch={false}
  />
  <SplitPane
    left={<Connect(FilmsFound) />}
    right={
      <SortBy
        onOptionChange={[MockFunction]}
        options={
          Array [
            Object {
              "id": "release_date",
              "label": "release date",
            },
            Object {
              "id": "vote_average",
              "label": "rating",
            },
          ]
        }
        selectedOptionId="release_date"
      />
    }
  />
  <PageContentWrapper>
    <Loader
      loaderColor="#ec407a"
      size="90px"
    />
  </PageContentWrapper>
  <Footer />
</PageGrid>
`;
